tools/py/committee-reweight.py

```diff
@@ -0,0 +1,194 @@
+    num_obs_models = obs.shape[1]
+
+    print("Computing h matrix")
+    # get the h matrix h = beta(V^i - barV)
+    h_matrix = np.zeros((num_pot_frames, num_pot_models))
+    # fast way avoiding double loop
+    h_matrix = beta * (pot.T - np.mean(pot, axis=1)).T
+    print("Computing averages")
+    # get frame-average of h, h_avg
+    h_avg = np.mean(h_matrix, axis=0)  # 1d-array of length num_pot_models
+    # get frame-average of observable, obs_avg
+    obs_avg = np.mean(obs, axis=0)  # 1d-array of length num_obs_models
+    # get frame-average of the product between the observable and h, obsxh_avg
+    obsxh_avg = (obs.T @ h_matrix) / num_pot_frames
+    # get frame-average of observable, reweighted according to the CEA, obs_avg_CEA
+    obs_avg_CEA = np.zeros((num_obs_models, num_pot_models))
+    for j in range(num_obs_models):
+        for i in range(num_pot_models):
+            obs_avg_CEA[j, i] = obs_avg[j] - obsxh_avg[j, i] + obs_avg[j] * h_avg[i]
+    return obs_avg_CEA, h_matrix
+
+
+def commitee_reweight(path2ixml, pot_file, obs_file, stride=1, index=-1, direct=False):
+    """
+    Parameters
+    ----------
+    pot_file    :   str, mandatory
+                    The file containing the values of the potential from each model, either in .dat form or as .extra.
+                    Each line contains the potentials for the M models for each frame
+    obs_file    :   str, mandatory
+                    A file containing the value of the observable for each frame.
+                    It is assumed that lines correspond to frames, while columns to properties.
+                    Multiple properties can be reweighted at the same time.
+    stride      :   integer, [1]
+                    The frequency of sampling of pot_file, if different from that of prop_file (e.g. --stride 10 if
+                    the potential was printed 10 times more often than the observable. --stride -10 must be used if
+                    the observable were printed more often than the potential).
+    index       :   integer, optional
+                    Which column of the property file should be used to compute the average and uncertainty.
+                    If it is not given, and there are multiple columns, they will be interpreted as corresponding
+                    to multiple property models
+    direct      :   bool, optional
+                    Activates the direct reweighting, instead of the cumulant expansion approximation.
+                    Use at your own risk!
+
+    """
+    if index >= 0:
+        obs = np.loadtxt(obs_file, usecols=index)
+    else:
+        obs = np.loadtxt(obs_file)
+
+
+    if stride > 0:
+        potentials = np.loadtxt(pot_file)[::stride]
+    elif stride < 0:
+        stride = np.abs(stride)
+        obs = obs[::stride]
+    else:
+        raise ValueError(
+                "Stride value cannot be zero"
+                )
+
+
+    # Load kbT from i-PI, we could make it into a small function
```

Comment:
does this handle different possible units for T and so on? I am assuming yes already in xml_parse. Comment here to check.

Reply:
Yes, the idea is that by pulling the temperature from the i-PI parser it takes care of the units of the temperature. The only thing to care about is feeding energies in Ha.